src/goto-programs/read_bin_goto_object.cpp

```diff
@@ -72,10 +72,16 @@ static bool read_bin_goto_object(
 
     if(!sym.is_type && sym.type.id()==ID_code)
     {
-      // makes sure there is an empty function
-      // for every function symbol and fixes
-      // the function types.
-      functions.function_map[sym.name].type=to_code_type(sym.type);
+      // makes sure there is an empty function for every function symbol
+      auto entry = functions.function_map.emplace(sym.name, goto_functiont());
+
+      const code_typet &code_type = to_code_type(sym.type);
+      entry.first->second.type = code_type;
+      entry.first->second.parameter_identifiers.reserve(
+        code_type.parameters().size());
+      for(const auto &parameter : code_type.parameters())
```

Comment:
:pick: :pick: I have a small preference for using `<algorithm>` (`std::transform` in this case) where applicable

Reply:
What would a concise piece of code achieving this look like? I'd like to have a list comprehension like `parameter_identifiers = [p.get_identifier() for p in code_type.parameters()];` - but that's Python, not C++...